pinot-query-planner/src/main/java/org/apache/pinot/query/rules/PinotJoinExchangeNodeInsertRule.java

```diff
@@ -70,17 +70,17 @@ public void onMatch(RelOptRuleCall call) {
     RelNode leftExchange;
     RelNode rightExchange;
     List<RelHint> hints = join.getHints();
-    if (hints.contains(PinotRelationalHints.USE_HASH_DISTRIBUTE)) {
+    if (hints.contains(PinotRelationalHints.USE_BROADCAST_DISTRIBUTE)) {
+      leftExchange = LogicalExchange.create(leftInput, RelDistributions.RANDOM_DISTRIBUTED);
+      rightExchange = LogicalExchange.create(rightInput, RelDistributions.BROADCAST_DISTRIBUTED);
```

Comment:
I don't follow this. If hint indicates to use BROADCAST_EXCHANGE, then why are we setting up exchanges for both sides of the JOIN ? In broadcast, only one side (either left or right) should be exchanged / moved, whereas other side remains as opposed to full shuffle based JOIN where data movement involves both sides.

Reply:
this is not true. for example if we were to leverage the full potential of the server instances, we need to randomly distribute the payload to all servers on left, but broadcast the right table to the same set of servers. 

limiting the left hand side to stay put meant we can't do a scale-up join operation which will take lots of memory/cpu

Reply:
I see. May be we can do that as a special optimization for BROADCAST exchange and not make it the default BROADCAST exchange behavior ?